### PR TITLE
Synchronize test-requirements.txt and .pre-commit-config.yaml

### PR DESCRIPTION
**Pull request by hamdanal**
The two files got out of sync after #15167

(Explain how this PR changes mypy.)

### CODE REVIEW

**Comment by hamdanal** [comment]
flake8 v6.0.0 doesn't support python 3.7 :/

**Comment by JelleZijlstra** [comment]
We should probably go back to flake8 5 in that case.

**Comment by AlexWaygood** [comment]
> We should probably go back to flake8 5 in that case.

And add a comment to `.pre-commit-config.yaml` saying why we're sticking on flake8 v5

**Comment by hamdanal** [comment]
OK isort says hi now :D

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -4,11 +4,11 @@ repos:
     hooks:
       - id: black
   - repo: https://github.com/pycqa/isort
-    rev: 5.12.0  # must match test-requirements.txt
+    rev: 5.11.5  # must match test-requirements.txt (cannot use version 2.12 until python 3.7 support is dropped)
```
Comment (AlexWaygood) [inline .pre-commit-config.yaml:7]: :)

```suggestion
    rev: 5.11.5  # must match test-requirements.txt (cannot use version 5.12 until python 3.7 support is dropped)
```

Comment (hamdanal) [reply .pre-commit-config.yaml]: Sorry pushed before your comment :/

**Comment by hamdanal** [comment]
It is probably better to run the linting CI on a newer version of python so that you don't have to deal with this situation anymore. Do you want me to make a PR or maybe update this one?

**Comment by AlexWaygood** [comment]
> It is probably better to run the linting CI on a newer version of python so that you don't have to deal with this situation anymore.

Yeah, just because we support Python 3.7 doesn't mean that we support the project being linted on Python 3.7. So I think that would be fine, and would make the situation easier

**Comment by AlexWaygood** [comment]
You could also update the pinned versions of flake8-bugbear and flake8-noqa. (There are no new hits from either of them.)

If we stick with flake8 v5, we can't upgrade to the latest version of flake8-bugbear (23.3.23), which requires flake8 v6, so we can only upgrade to flake8-bugbear==23.3.12. flake8-noqa can be upgraded to the latest version (1.3.1).

**Comment by hamdanal** [comment]
> > It is probably better to run the linting CI on a newer version of python so that you don't have to deal with this situation anymore.
> 
> Yeah, just because we support Python 3.7 doesn't mean that we support the project being linted on Python 3.7. So I think that would be fine, and would make the situation easier

Do you propose that I change it in this PR or a separate one?

**Comment by AlexWaygood** [comment]
> Do you propose that I change it in this PR or a separate one?

I guess a separate PR makes more sense, but no strong opinion either way

**Pull request merged**